testflinger_cli/__init__.py

```diff
@@ -451,10 +454,12 @@ def do_poll(self, job_id):
         prev_queue_pos = None
         if job_state == "waiting":
             print("This job is waiting on a node to become available.")
-        while job_state != "complete":
-            if job_state == "cancelled":
-                break
+        while True:
```

Comment:
Why do you use `while True` here? Maybe it shouldn’t be an infinite loop, but we can try to execute the next code a finite number of times before giving up?

Reply:
Yes, that was an intentional change here to simplify things a bit. We were already checking the state to break out of the loop if the job gets cancelled. Now we just do the check and also break out if it goes to complete. If it's being run interactively, then using ^C can still be used to manually exit the run before it completes. It could also be cancelled or exited in a similar way by jenkins job timeout, or by going to a cancelled state from testflinger because it takes too long.